File: 594-longest-harmonious-subsequence/longest-harmonious-subsequence.go
func findLHS(nums []int) int {
    sort.Ints(nums)
    j, maxlen := 0, 0
    for i:=0; i<len(nums); i++ {
        if nums[i] - nums[j] == 1 {
            maxlen = max(maxlen, (i - j + 1))
        }
        for nums[i] - nums[j] > 1 {
            j++
        }
    }

    return maxlen
}

File: 594-longest-harmonious-subsequence/longest-harmonious-subsequence.cpp
class Solution {
public:
    int findLHS(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int i = 0, len = 0;
        for(int j=0; j<nums.size(); j++){
            if (nums[j] - nums[i] == 1){
                len = max(len, (j-i+1));
            }
            while (nums[j] - nums[i] > 1) {
                i++;
            }
        }
        return len;
    }
};